### PR TITLE
feat: improve the error handling in sql cluster router

### PR DESCRIPTION
**Pull request by vagetablechicken**
ref #2741

### CODE REVIEW

File: hybridse/include/sdk/base.h [hunk 1]
```diff
@@ -31,14 +31,38 @@
...
+    }
+
+    std::string ToString() const {
+        std::string str("[");
+        str.append(std::to_string(code)).append("] ").append(msg);
+        return str;
+    };
+
     int code;
     std::string msg;
```
Comment (aceforeverd) [inline hybridse/include/sdk/base.h:78]: memory efficiency with `absl::Cord`  ?

File: src/base/status_util.h [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+/// @brief If the status is bad, CHECK immediately, appending the status to the
+///   logged message.
+#define CHECK_OK(s) CHECK_OK_PREPEND(s, "Bad status")
+
+/// @brief If @c to_call returns a bad status, DCHECK immediately with
+///   a logged message of @c msg followed by the status.
+#define DCHECK_OK_PREPEND(to_call, msg)                \
+    do {                                               \
+        const ::hybridse::sdk::Status& _s = (to_call); \
+        DCHECK(_s.IsOK()) << (msg) << ": " << _s.msg;  \
```
Comment (aceforeverd) [inline src/base/status_util.h:218]: only check in debug mode ?

File: hybridse/include/sdk/base.h [hunk 3]
```diff
@@ -31,14 +31,38 @@
...
         : code(status_code), msg(msg_str), trace(trace) {}
     bool IsOK() const { return code == 0; }
 
+    void SetOK() {
+        code = 0;
+        msg = "ok";
+    }
+    void SetCode(int c) { code = c; }
+    void SetMsg(const std::string& new_msg) { msg = new_msg; }
+    void SetTraces(const std::string& new_trace) { trace = new_trace; }
```
Comment (aceforeverd) [inline hybridse/include/sdk/base.h]: `string_view` might a little better

**Review by dl239 — APPROVED** [review]

File: src/base/status_util.h (hunk 2 above)
Comment (vagetablechicken) [reply src/base/status_util.h:218]: yes, just a debug macro, unused now

File: hybridse/include/sdk/base.h (hunk 1 above)
Comment (vagetablechicken) [reply hybridse/include/sdk/base.h:78]: we may use status.msg as a std::string, can't convert it to Cord now

File: hybridse/include/sdk/base.h (hunk 3 above)
Comment (vagetablechicken) [reply hybridse/include/sdk/base.h]: ok

Comment (vagetablechicken) [reply hybridse/include/sdk/base.h]: ok

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**